packages/moonstone/Scroller/Scroller.js

```diff
@@ -152,98 +151,69 @@ class ScrollerBase extends Component {
+			if (isItemBeforeView(ib, sb, d)) {
+				return roundToBoundary(sb, calcItemAtStart(ib, sb, st, d), sh, d);
+			} else if (isItemAfterView(ib, sb, d)) {
+				return roundToBoundary(sb, calcItemAtEnd(ib, sb, st, d), sh, d);
 			}
-		}
 
-		if (itemHeight > clientHeight) {
-			// Calculations for `containerHeight` that are bigger than `clientHeight`
-			const
-				{top, height: nestedItemHeight} = focusedItem.getBoundingClientRect(),
-				nestedItemTop = this.uiRef.containerRef.scrollTop + (top - containerTop),
-				nestedItemBottom = nestedItemTop + nestedItemHeight;
-
-			if (nestedItemBottom - scrollBottom > epsilon) {
-				// Calculate when 5-way focus down past the bottom.
-				newScrollTop += nestedItemBottom - scrollBottom;
-			} else if (nestedItemTop - currentScrollTop < epsilon) {
-				// Calculate when 5-way focus up past the top.
-				if (newItemTop > newScrollTop) {
-					// Ensure that the adjusted scrollTop would at least scroll the container to the top of
-					// the viewport (e.g. because the container is at the bottom of the scroller and the
-					// nested item is at the top of the container)
-					newScrollTop = newItemTop;
-				} else {
-					newScrollTop += nestedItemTop - currentScrollTop;
-				}
-			} else if (newItemTop - nestedItemHeight - currentScrollTop > epsilon) {
-				// set scroll position so that the top of the container is at least on the top as a fallback.
-				newScrollTop = newItemTop - nestedItemHeight;
-			}
-		} else if (itemBottom - scrollBottom > epsilon) {
-			// Calculate when 5-way focus down past the bottom.
+			return st;
+		};
 
-			// if the last item is focused and have an invisible area, scroll all the way to the bottom
-			if (scrollHeight - itemBottom < heightThreshold) {
-				newScrollTop += scrollHeight - scrollBottom;
-			} else {
-				newScrollTop += itemBottom - scrollBottom;
+		const container = this.getSpotlightContainerForNode(item);
+		const scrollerBounds = this.uiRef.containerRef.getBoundingClientRect();
+		let {scrollHeight, scrollTop} = this.uiRef.containerRef;
+		let scrollTopDelta = 0;
+
+		const adjustScrollTop = (v) => {
+			scrollTopDelta = scrollTop - v;
+			scrollTop = v;
+		};
+
+		if (container) {
+			const containerBounds = container.getBoundingClientRect();
+
+			// if the entire container fits in the scroller, scroll it into view
+			if (canItemFit(containerBounds, scrollerBounds)) {
+				return calcItemInView(containerBounds, scrollerBounds, scrollTop, scrollHeight, scrollTopDelta);
 			}
-		} else if (newItemTop - currentScrollTop < epsilon && scrollHeightChange <= 0) {
-			// Calculate when 5-way focus up past the top.
 
-			// if the first item is focused and have an invisible area, scroll all the way to the top
-			if (newItemTop < heightThreshold) {
-				newScrollTop = 0;
-			} else {
-				newScrollTop += newItemTop - currentScrollTop;
+			// if the container doesn't fit, adjust the scroll top ...
+			if (containerBounds.top > scrollerBounds.top) {
```

Comment:
I'm not sure but don't we need add the following condition in the as-is code to the above condition because it seems that the top of the container needs to stick to the top of the scroller when only the focused item bottom is less then scrollTop.
```
if (nestedItemTop - currentScrollTop < epsilon)
```

If not, for example,
When opening the 2nd ExpandableList in qa-scroller sample, both of `containerBounds.top > scrollerBounds.top` and `containerBounds.top + containerBounds.height < scrollerBounds.top + scrollerBounds.height` conditions are true, so finally the bottom of the container sticks to the boom of the scroller instead of the top of the scroller even though `containerBounds.top > scrollerBounds.top`  condition is true.

Reply:
I'm not following what you expect to see. Right now, it's set up to anchor the container to the top of the viewport if the container is below the top of the viewport and the bottom if the bottom of the container is above the bottom of the viewport.

Does that match your expectation? Can you share more specific steps to repro the problem?

Reply:
When trying to reproduce this issue with the latest code, I could not reproduce it.